lib/Core/loadPlugins.ts

```diff
@@ -0,0 +1,34 @@
+import {
+  TerriaPlugin,
+  createPluginContext,
+  ViewState
+} from "terriajs-plugin-api";
+
+type PluginModule = { default: TerriaPlugin };
+
+/**
+ * Load plugin modules.
+ *
+ * @param pluginPromises Array of promise returning plugin modules.
+ */
+async function loadPlugins(
+  viewState: ViewState,
+  pluginPromises: Promise<PluginModule>[]
+): Promise<void> {
+  const loadPromises = pluginPromises.map(promise => {
+    const pluginContext = createPluginContext(viewState);
+    promise
+      .then(({ default: plugin }) => {
+        try {
+          plugin.register(pluginContext);
+        } catch (error) {
+          console.error(`Error when registering plugin "${plugin.name}"`);
```

Comment:
For the sake of consistency, should we be using a TerriaError here?

Reply:
Good point. I'll check how to use it here.

Reply:
At this point we don't want to report the error to the user, so a console.error will be enough I think. I have still converted the error to `TerriaError` before logging to console.